File: src/main/java/ch/hevs/sdi/simpleapp/SimpleappConfiguration.java
package ch.hevs.sdi.simpleapp;

import org.springframework.context.annotation.*;

@Configuration
public class SimpleappConfiguration {
    @Bean
    @Profile("default")
    Printer simplePrinter() {
        return new SimplePrinter();
    }

    @Bean
    @Profile("fancy")
    Printer fancyPrinter() {
        return new FancyPrinter();
    }
}

File: src/main/java/ch/hevs/sdi/simpleapp/SimpleappConfiguration.kt
package ch.hevs.sdi.simpleapp

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Profile

@Configuration
open class SimpleappConfiguration {
    @Bean
    @Profile("default")
    open fun printer() = SimplePrinter()

    @Bean
    @Profile("fancy")
    open fun fancyPrinter() = FancyPrinter()
}
